tests/ui/traits/next-solver/more-object-bound.stderr

```diff
@@ -1,17 +1,9 @@
-error[E0271]: type mismatch resolving `<dyn Trait<A = A, B = B> as SuperTrait>::A == B`
+error[E0271]: type mismatch resolving `A == B`
```

Comment:
`AliasRelate` goals are negatively affected by this PR, which should make sense b/c we're normalizing the LHS and RHS; we probably should special-case `ProjectionPredicate`s, since handling `AliasRelate` predicates is already kinda weird: https://github.com/rust-lang/rust/blob/d4f880f8ce832cd7560bb2f1ebc34f967055ffd7/compiler/rustc_trait_selection/src/error_reporting/traits/fulfillment_errors.rs#L1422-L1465.

Reply:
hmm yeah, I feel like we should do the "alias-relate -> normalization failure" transformation in the `BestObligation` visitor instead maybe :thinking: 

This is definitely worse... also in the `next-solver/async.rs#fail` test

Reply:
I'd rather defer this to a different PR. Right now the regression seems like a logical consequence of this implementation, and I'll play around with a fix for the `AliasRelate` regression separately.